### PR TITLE
Add tool to clean up timed out builds

### PR DESCRIPTION
**Pull request by mr0re1**
```shell
$ tools/cloud-build/purge_timedout.py 
[0]     2023-05-01      https://console.cloud.google.com/cloud-build/builds/X?project=X   DAILY-test-slurm-gcp-v5-ubuntu2004
[1]     2023-05-01      https://console.cloud.google.com/cloud-build/builds/X?project=X   DAILY-test-slurm-gcp-v5-startup-scripts
[2]     2023-05-01      https://console.cloud.google.com/cloud-build/builds/X?project=X   DAILY-test-slurm-gcp-v5-hpc-centos7
[3]     2023-05-01      https://console.cloud.google.com/cloud-build/builds/X?project=X   DAILY-test-hpc-high-io-v5
Select build # to purge:
```

### CODE REVIEW

File: tools/cloud-build/purge_timedout.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    logs = bucket.blob("log-%s.txt" % build.id).download_as_string().decode('utf-8')
+    mtch = re.search("gs://[\w-]*tf-state/.*\.tgz", logs)
+    assert mtch is not None, "can find link to tfstate"
+    link = mtch.group()
+    res = os.system(f"""gcloud storage cp {link} "/tmp/to_purge_tgz" && \
+rm -rf "/tmp/to_purge"; mkdir "/tmp/to_purge" && \
+tar zxf "/tmp/to_purge_tgz" -C "/tmp/to_purge" """)
+    assert res == 0
+    for d in glob.glob("/tmp/to_purge/*/*"):
+        res = os.system(f"terraform -chdir={d} init && terraform -chdir={d} destroy --auto-approve")
```
Comment (tpdownes) [inline tools/cloud-build/purge_timedout.py:61]: Single dash on `-auto-approve`

File: tools/cloud-build/purge_timedout.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        project_id=project, 
+        filter="status=TIMEOUT", 
+        page_size=1000
+    )
+    builds = cb.list_builds(req).builds
+    if not builds:
+        print("No timedout builds found this week.")
+        return
+
+    PDT = timezone(timedelta(hours=-7))
```
Comment (tpdownes) [inline tools/cloud-build/purge_timedout.py:38]: I don't see what value setting the timezone to non-UTC is getting us here.

File: tools/cloud-build/purge_timedout.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    pastweek = datetime.combine(date.today(), datetime.min.time()).astimezone(PDT) - timedelta(days=7)
```
Comment (tpdownes) [inline tools/cloud-build/purge_timedout.py:39]: I think `datetime.combine(date.today()...).astimezone(PDT)` only does the right thing if run from within the PDT (which is only active half the year)

File: tools/cloud-build/purge_timedout.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    res = os.system(f"""gcloud storage cp {link} "/tmp/to_purge_tgz" && \
+rm -rf "/tmp/to_purge"; mkdir "/tmp/to_purge" && \
+tar zxf "/tmp/to_purge_tgz" -C "/tmp/to_purge" """)
+    assert res == 0
+    for d in glob.glob("/tmp/to_purge/*/*"):
+        res = os.system(f"terraform -chdir={d} init && terraform -chdir={d} destroy --auto-approve")
+        assert res == 0
+    
+
+def get_default_project():
```
Comment (tpdownes) [inline tools/cloud-build/purge_timedout.py:65]: in what context is this code expected to run? Cloud Build/Functions/Run/etc? There may be smarter ways of determining the project.

**Review by tpdownes — CHANGES_REQUESTED** [review]

File: tools/cloud-build/purge_timedout.py (hunk 4 above)
Comment (mr0re1) [reply tools/cloud-build/purge_timedout.py:65]: Intention is to run it in dev machine terminal.

**Pull request closed without merging**